apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java

```diff
@@ -398,8 +398,18 @@ private <T> List<T> findClaimTypeByPatient(
     criteria.select(root).distinct(true);
 
     // Search for a beneficiary's records. Use lastUpdated if present
+    // TODO - BFD-1596
+    // while we gradually convert entity beans to use long data type for patientId,
+    // we may need to change the expected type for beneficiaryId in the entity Predicate.
+    // Once all claims have been migrated we can modify the ClaimTypeV2 to specifically
+    // return a long data type and this extra data type checking can be removed.
+    java.lang.Class javaClass = claimType.getEntityBeneficiaryIdAttribute().getJavaType();
+
     Predicate wherePredicate =
-        builder.equal(root.get(claimType.getEntityBeneficiaryIdAttribute()), patientId);
+        builder.equal(
+            root.get(claimType.getEntityBeneficiaryIdAttribute()),
+            javaClass.getName().equals("long") ? Long.parseLong(patientId) : patientId);
```

Comment:
I think getJavaType was returning class.getName under certain conditions, wasnt it? Will that return "Long" or the lowercase version? (And if it returns the same value, why do we return the same value in two different ways within getJavaType?)

Reply:
Depends on the actual db column type AND whether the db column can be nullable. There is a very simple rule to be applied here:

If the db column is defined as an integral data data **AND that column cannot be null**, then use a native primitive; in this case 99% of the time, beneficiaryId is not nullable and therefore ORM models need to treat it as a long, period.

However, we have one table, medicare_benficiaryid_history where bene_id is defined as nullable; so for that we need a Long data type.

Reply:
Thanks for context on the temp migration code! Looking forward to this being simplified once we're done with the migrations. (Maybe the temporary code should be denoted as such for easier removal later?)